cmake/compiler_settings.cmake

```diff
@@ -76,9 +78,12 @@ if (CMAKE_CXX_COMPILER_ID MATCHES GNU OR CMAKE_CXX_COMPILER_ID MATCHES Clang)
   # to fix warnings as they arise, so they don't accumulate "to be fixed later".
   add_compile_options(-Wall -Werror)
 
-  add_compile_flags("C;CXX" -fno-strict-aliasing)
+  add_compile_options(-fno-strict-aliasing)
 
-  add_compile_flags_if_supported(C -Wjump-misses-init)
+  check_c_compiler_flag(-Wjump-misses-init WARN_JUMP_MISSES_INIT_SUPPORTED)
```

Comment:
@mikbras should this be globally defined? It already was, so I left it as such, but in general global side effects in CMake are not great. If you know which targets this should be specifically applied to... let me know!

Reply:
I would restrict this to core, enclave, and host directories.

Reply:
Thank you!

Reply:
-Wc++-compat added too many extra warnings. We probably _should_ add it, but not now. So I've just added -Wjump-misses-init to said targets, when using GCC (as it's not supported with Clang).